### ISSUE TITLE
Overflow value is inlined and takes precedence over any class

### ISSUE THREAD

**Issue by wiwofone** [angular/material#2185]
Currently, when a dialog opens and given that the parent scroll option is enabled (which it is by default), the previous `overflow` value of the parent element is saved, and `overflow` is set to hidden. When the dialog is closed, the value is set back to the previous value that was saved.

As taken from the source, onRemove does this:

    if (options.disableParentScroll) {
        options.parent.css('overflow', options.lastOverflow);
        delete options.lastOverflow;
    }

This presents a problem when the overflow value of the parent element is already set. Specifically, if I have a class on the parent element that already has `overflow: hidden`, that value will be saved. When the dialog is closed, `lastOverflow` will be `hidden`, and the parent element will get `overflow: hidden` in the inline style. This takes precedence over any classes, which means that when I later remove the aforementioned class, the parent will still be unscrollable.

TL;DR:
1. Set `overflow: hidden` to a dialog parent by adding a class (e.g. `.unscrollable`).
2. Open a dialog
3. Close the dialog
4. Remove the `.unscrollable` class.
5. The parent is still unscrollable since ngMaterial inlined `overflow: hidden`.

**Comment by princemaple** [comment]
This also causes serious performance issue, namely a total dom reflow on the container. When there are many elements in the container, the adding/removing of the scrollbar triggers the reflow and the UI can get stuck.

**Comment by Luxiyalu** [comment]
Several appearances of these in the whole system, some interfering with one another, e.g. $mdDialog with $mdSidenav. They should be: `options.parent.css('overflow', '');`, **removing** those inline value, not card-code them to their original state yet again.

**Comment by rschmukler** [comment]
This should now be fixed since `$mdDialog` now uses `$mdUtil.disableScrollAround` which is more responsible with how it cleans up styles.
